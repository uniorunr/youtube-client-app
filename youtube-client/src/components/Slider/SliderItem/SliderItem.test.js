import sliderItemComponent from './SliderItem';

describe('sliderItemComponent', () => {
  it('should be instance of Function', () => {
    expect(sliderItemComponent).toBeInstanceOf(Function);
  });

  it('should be rendered correctly if two arguments passed', () => {
    const data = {
      testVideoId1: {
        channelTitle: 'channelTitle1',
        description: 'description1',
        publishedAt: 'publishedAt1',
        thumbnailUrl: 'thumbnailUrl1',
        title: 'title1',
        viewCount: 'viewCount1',
      },
      testVideoId2: {
        channelTitle: 'channelTitle2',
        description: 'description2',
        publishedAt: 'publishedAt2',
        thumbnailUrl: 'thumbnailUrl2',
        title: 'title2',
        viewCount: 'viewCount2',
      },
    };
    sliderItemComponent('body', data);
    expect(document.body.innerHTML).toMatchSnapshot();
    document.body.innerHTML = '';
  });

  it('should be rendered correctly if three arguments passed', () => {
    const data = {
      testVideoId1: {
        channelTitle: 'channelTitle1',
        description: 'description1',
        publishedAt: 'publishedAt1',
        thumbnailUrl: 'thumbnailUrl1',
        title: 'title1',
        viewCount: 'viewCount1',
      },
      testVideoId2: {
        channelTitle: 'channelTitle2',
        description: 'description2',
        publishedAt: 'publishedAt2',
        thumbnailUrl: 'thumbnailUrl2',
        title: 'title2',
        viewCount: 'viewCount2',
      },
    };
    sliderItemComponent('body', data, 5);
    expect(document.body.innerHTML).toMatchSnapshot();
    document.body.innerHTML = '';
  });
});
