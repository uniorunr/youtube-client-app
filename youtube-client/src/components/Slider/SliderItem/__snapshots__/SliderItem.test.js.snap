// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sliderItemComponent should be rendered correctly if three arguments passed 1`] = `
"<article class=\\"container container5\\">
<div class=\\"youtube-img-wrapper\\">
  <img class=\\"youtube-img img5\\" src=\\"thumbnailUrl1\\" alt=\\"Youtube preview image\\">
</div>
<div class=\\"container-content\\">
  <header class=\\"container-header\\">
    <h2 class=\\"container-header-title\\">
      <a class=\\"container-header-title-link title-link5\\" href=\\"https://www.youtube.com/watch?v=testVideoId1\\" target=\\"_blank\\">title1</a>
    </h2>
    <span class=\\"container-header-channel channel5\\">channelTitle1</span>
  </header>
  <section class=\\"container-description description5\\">
    <p>description1</p>
  </section>
  <footer class=\\"container-footer\\">
    <div class=\\"container-footer-wrapper\\">
      <div class=\\"container-footer-reg\\">
        <i class=\\"far fa-calendar-alt\\"></i>
        <span class=\\"container-footer-date date5\\">publishedAt1</span>
      </div>
      <div class=\\"container-footer-view-sec\\">
        <i class=\\"far fa-eye\\"></i>
        <span class=\\"container-footer-views views5\\">viewCount1</span>
      </div>
    </div>
   </footer>
</div>
</article><article class=\\"container container6\\">
<div class=\\"youtube-img-wrapper\\">
  <img class=\\"youtube-img img6\\" src=\\"thumbnailUrl2\\" alt=\\"Youtube preview image\\">
</div>
<div class=\\"container-content\\">
  <header class=\\"container-header\\">
    <h2 class=\\"container-header-title\\">
      <a class=\\"container-header-title-link title-link6\\" href=\\"https://www.youtube.com/watch?v=testVideoId2\\" target=\\"_blank\\">title2</a>
    </h2>
    <span class=\\"container-header-channel channel6\\">channelTitle2</span>
  </header>
  <section class=\\"container-description description6\\">
    <p>description2</p>
  </section>
  <footer class=\\"container-footer\\">
    <div class=\\"container-footer-wrapper\\">
      <div class=\\"container-footer-reg\\">
        <i class=\\"far fa-calendar-alt\\"></i>
        <span class=\\"container-footer-date date6\\">publishedAt2</span>
      </div>
      <div class=\\"container-footer-view-sec\\">
        <i class=\\"far fa-eye\\"></i>
        <span class=\\"container-footer-views views6\\">viewCount2</span>
      </div>
    </div>
   </footer>
</div>
</article>"
`;

exports[`sliderItemComponent should be rendered correctly if two arguments passed 1`] = `
"<article class=\\"container container0\\">
<div class=\\"youtube-img-wrapper\\">
  <img class=\\"youtube-img img0\\" src=\\"thumbnailUrl1\\" alt=\\"Youtube preview image\\">
</div>
<div class=\\"container-content\\">
  <header class=\\"container-header\\">
    <h2 class=\\"container-header-title\\">
      <a class=\\"container-header-title-link title-link0\\" href=\\"https://www.youtube.com/watch?v=testVideoId1\\" target=\\"_blank\\">title1</a>
    </h2>
    <span class=\\"container-header-channel channel0\\">channelTitle1</span>
  </header>
  <section class=\\"container-description description0\\">
    <p>description1</p>
  </section>
  <footer class=\\"container-footer\\">
    <div class=\\"container-footer-wrapper\\">
      <div class=\\"container-footer-reg\\">
        <i class=\\"far fa-calendar-alt\\"></i>
        <span class=\\"container-footer-date date0\\">publishedAt1</span>
      </div>
      <div class=\\"container-footer-view-sec\\">
        <i class=\\"far fa-eye\\"></i>
        <span class=\\"container-footer-views views0\\">viewCount1</span>
      </div>
    </div>
   </footer>
</div>
</article><article class=\\"container container1\\">
<div class=\\"youtube-img-wrapper\\">
  <img class=\\"youtube-img img1\\" src=\\"thumbnailUrl2\\" alt=\\"Youtube preview image\\">
</div>
<div class=\\"container-content\\">
  <header class=\\"container-header\\">
    <h2 class=\\"container-header-title\\">
      <a class=\\"container-header-title-link title-link1\\" href=\\"https://www.youtube.com/watch?v=testVideoId2\\" target=\\"_blank\\">title2</a>
    </h2>
    <span class=\\"container-header-channel channel1\\">channelTitle2</span>
  </header>
  <section class=\\"container-description description1\\">
    <p>description2</p>
  </section>
  <footer class=\\"container-footer\\">
    <div class=\\"container-footer-wrapper\\">
      <div class=\\"container-footer-reg\\">
        <i class=\\"far fa-calendar-alt\\"></i>
        <span class=\\"container-footer-date date1\\">publishedAt2</span>
      </div>
      <div class=\\"container-footer-view-sec\\">
        <i class=\\"far fa-eye\\"></i>
        <span class=\\"container-footer-views views1\\">viewCount2</span>
      </div>
    </div>
   </footer>
</div>
</article>"
`;
